### ISSUE TITLE
Add support for `has_projects`

### ISSUE THREAD

**Issue by patcon** [probot/settings#34]
Didn't seem to work here: https://github.com/edgi-govdata-archiving/edgi-scripts/commit/b311b73e87398f6c38f9f71bc332f2a0ac286039

**Comment by hiimbex** [comment]
Hi @patcon ! I was also able to reproduce this issue here:
```
C02TR0KFFVH4:configurer github$ curl -v -XPATCH "https://api.github.com/repos/hiimbex-test/test" -H "Authorization: token $MY_TOKEN" -H "Content-Type: application/json" --data '{
> "has_project": false,
> "name": "test"
> }'
*   Trying 192.30.253.117...
* TCP_NODELAY set
* Connected to api.github.com (192.30.253.117) port 443 (#0)
* TLS 1.2 connection using TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256
* Server certificate: *.github.com
* Server certificate: DigiCert SHA2 High Assurance Server CA
* Server certificate: DigiCert High Assurance EV Root CA
> PATCH /repos/hiimbex-test/test HTTP/1.1
> Host: api.github.com
> User-Agent: curl/7.51.0
> Accept: */*
> Authorization: token $MY_TOKEN
> Content-Type: application/json
> Content-Length: 40
> 
* upload completely sent off: 40 out of 40 bytes
< HTTP/1.1 200 OK
< Server: GitHub.com
< Date: Fri, 14 Jul 2017 02:49:37 GMT
< Content-Type: application/json; charset=utf-8
< Content-Length: 5215
< Status: 200 OK
< X-RateLimit-Limit: 5000
< X-RateLimit-Remaining: 4987
< X-RateLimit-Reset: 1500003033
< Cache-Control: private, max-age=60, s-maxage=60
< Vary: Accept, Authorization, Cookie, X-GitHub-OTP
< ETag: "b1f4783a4f9e1f0d5319994426026bec"
< X-OAuth-Scopes: admin:gpg_key, admin:org, admin:org_hook, admin:public_key, admin:repo_hook, delete_repo, gist, notifications, repo, user
< X-Accepted-OAuth-Scopes: 
< X-GitHub-Media-Type: github.v3; format=json
< Access-Control-Expose-Headers: ETag, Link, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval
< Access-Control-Allow-Origin: *
< Content-Security-Policy: default-src 'none'
< Strict-Transport-Security: max-age=31536000; includeSubdomains; preload
< X-Content-Type-Options: nosniff
< X-Frame-Options: deny
< X-XSS-Protection: 1; mode=block
< X-Runtime-rack: 0.058427
< X-GitHub-Request-Id: EA19:5F1B:706060:C47604:59683140
< 
{
  "id": 97184633,
  "name": "test",
  "full_name": "hiimbex-test/test",
  "owner": {
    "login": "hiimbex-test",
    "id": 29493459,
    "avatar_url": "https://avatars5.githubusercontent.com/u/29493459?v=4",
    "gravatar_id": "",
    "url": "https://api.github.com/users/hiimbex-test",
    "html_url": "https://github.com/hiimbex-test",
    "followers_url": "https://api.github.com/users/hiimbex-test/followers",
    "following_url": "https://api.github.com/users/hiimbex-test/following{/other_user}",
    "gists_url": "https://api.github.com/users/hiimbex-test/gists{/gist_id}",
    "starred_url": "https://api.github.com/users/hiimbex-test/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/hiimbex-test/subscriptions",
    "organizations_url": "https://api.github.com/users/hiimbex-test/orgs",
    "repos_url": "https://api.github.com/users/hiimbex-test/repos",
    "events_url": "https://api.github.com/users/hiimbex-test/events{/privacy}",
    "received_events_url": "https://api.github.com/users/hiimbex-test/received_events",
    "type": "User",
    "site_admin": false
  },
  "private": false,
  "html_url": "https://github.com/hiimbex-test/test",
  "description": null,
  "fork": false,
  "url": "https://api.github.com/repos/hiimbex-test/test",
  "forks_url": "https://api.github.com/repos/hiimbex-test/test/forks",
  "keys_url": "https://api.github.com/repos/hiimbex-test/test/keys{/key_id}",
  "collaborators_url": "https://api.github.com/repos/hiimbex-test/test/collaborators{/collaborator}",
  "teams_url": "https://api.github.com/repos/hiimbex-test/test/teams",
  "hooks_url": "https://api.github.com/repos/hiimbex-test/test/hooks",
  "issue_events_url": "https://api.github.com/repos/hiimbex-test/test/issues/events{/number}",
  "events_url": "https://api.github.com/repos/hiimbex-test/test/events",
  "assignees_url": "https://api.github.com/repos/hiimbex-test/test/assignees{/user}",
  "branches_url": "https://api.github.com/repos/hiimbex-test/test/branches{/branch}",
  "tags_url": "https://api.github.com/repos/hiimbex-test/test/tags",
  "blobs_url": "https://api.github.com/repos/hiimbex-test/test/git/blobs{/sha}",
  "git_tags_url": "https://api.github.com/repos/hiimbex-test/test/git/tags{/sha}",
  "git_refs_url": "https://api.github.com/repos/hiimbex-test/test/git/refs{/sha}",
  "trees_url": "https://api.github.com/repos/hiimbex-test/test/git/trees{/sha}",
  "statuses_url": "https://api.github.com/repos/hiimbex-test/test/statuses/{sha}",
  "languages_url": "https://api.github.com/repos/hiimbex-test/test/languages",
  "stargazers_url": "https://api.github.com/repos/hiimbex-test/test/stargazers",
  "contributors_url": "https://api.github.com/repos/hiimbex-test/test/contributors",
  "subscribers_url": "https://api.github.com/repos/hiimbex-test/test/subscribers",
  "subscription_url": "https://api.github.com/repos/hiimbex-test/test/subscription",
  "commits_url": "https://api.github.com/repos/hiimbex-test/test/commits{/sha}",
  "git_commits_url": "https://api.github.com/repos/hiimbex-test/test/git/commits{/sha}",
  "comments_url": "https://api.github.com/repos/hiimbex-test/test/comments{/number}",
  "issue_comment_url": "https://api.github.com/repos/hiimbex-test/test/issues/comments{/number}",
  "contents_url": "https://api.github.com/repos/hiimbex-test/test/contents/{+path}",
  "compare_url": "https://api.github.com/repos/hiimbex-test/test/compare/{base}...{head}",
  "merges_url": "https://api.github.com/repos/hiimbex-test/test/merges",
  "archive_url": "https://api.github.com/repos/hiimbex-test/test/{archive_format}{/ref}",
  "downloads_url": "https://api.github.com/repos/hiimbex-test/test/downloads",
  "issues_url": "https://api.github.com/repos/hiimbex-test/test/issues{/number}",
  "pulls_url": "https://api.github.com/repos/hiimbex-test/test/pulls{/number}",
  "milestones_url": "https://api.github.com/repos/hiimbex-test/test/milestones{/number}",
  "notifications_url": "https://api.github.com/repos/hiimbex-test/test/notifications{?since,all,participating}",
  "labels_url": "https://api.github.com/repos/hiimbex-test/test/labels{/name}",
  "releases_url": "https://api.github.com/repos/hiimbex-test/test/releases{/id}",
  "deployments_url": "https://api.github.com/repos/hiimbex-test/test/deployments",
  "created_at": "2017-07-14T02:32:26Z",
  "updated_at": "2017-07-14T02:32:26Z",
  "pushed_at": "2017-07-14T02:32:26Z",
  "git_url": "git://github.com/hiimbex-test/test.git",
  "ssh_url": "git@github.com:hiimbex-test/test.git",
  "clone_url": "https://github.com/hiimbex-test/test.git",
  "svn_url": "https://github.com/hiimbex-test/test",
  "homepage": null,
  "size": 0,
  "stargazers_count": 0,
  "watchers_count": 0,
  "language": null,
  "has_issues": true,
  "has_projects": true,
  "has_downloads": true,
  "has_wiki": true,
  "has_pages": false,
  "forks_count": 0,
  "mirror_url": null,
  "open_issues_count": 0,
  "forks": 0,
  "open_issues": 0,
  "watchers": 0,
  "default_branch": "master",
  "permissions": {
    "admin": true,
    "push": true,
    "pull": true
  },
  "allow_squash_merge": true,
  "allow_merge_commit": true,
  "allow_rebase_merge": true,
  "network_count": 0,
  "subscribers_count": 0
}
* Curl_http_done: called premature == 0
* Connection #0 to host api.github.com left intact
```

cc/ @bkeepers

**Comment by bkeepers** [comment]
> I was also able to reproduce this issue…`"has_project": false`

@hiimbex Good idea using the curl command. I used that as a starting point, and while fiddling around with it noticed your original command used `has_project` (singular), but it should be `has_projects` (plural). This worked for me:

```
$ curl -i -XPATCH "https://api.github.com/repos/robotland/test" \
       -H "Authorization: token $(gh-token)" \
       -H "Content-Type: application/json" \
       --data '{"name": "test", "has_projects": true}' | grep has_projects

…

  "has_projects": false,
```

After confirming that was working, I looked at [node-github](https://github.com/mikedeboer/node-github) and saw that it was removing the `has_projects` param from the request. I opened https://github.com/mikedeboer/node-github/pull/556 to add support for this to the client.
